### ISSUE TITLE
Java warnings

### ISSUE THREAD

**Issue by rhc54** [open-mpi/ompi#2040]
I'm getting a ton of warnings when building the Java bindings:

```
./Intracomm.java:589: warning: no @throws for mpi.MPIException
public static String openPort() throws MPIException
                     ^
```

I can't seem to find a way to turn these off - any suggestions as to the source of the problem?

**Comment by hppritcha** [comment]
what version of javac are you using?

**Comment by nrgraham23** [comment]
What is your configure line?  I do not get these errors.

**Comment by nrgraham23** [comment]
Also, is this for master or the 1.10 series?  Because we fixed an issue where we were getting a ton of warnings on the java bindings, but I don't recall if it was sent to 1.10 too.

**Comment by rhc54** [comment]
This is indeed 1.10, so that may be the problem. Is it possible to get a PR for that change?

Here is how I configure:
```
enable_orcm=no
enable_oshmem=yes
enable_sensors=no
enable_orterun_prefix_by_default=yes
enable_mpi_thread_multiple=no
enable_mem_debug=no
enable_mem_profile=no
enable_debug_symbols=yes
enable_binaries=yes
enable_heterogeneous=no
enable_picky=yes
enable_debug=yes
enable_shared=yes
enable_static=no
enable_memchecker=no
enable_ipv6=yes
enable_mpi_fortran=yes
enable_mpi_cxx=no
enable_mpi_cxx_seek=no
enable_cxx_exceptions=no
enable_mpi_java=yes
enable_io_romio=no
enable_vt=no
enable_contrib_no_build=libnbc,vt
enable_mca_no_build=btl-usnic
with_memory_manager=no
with_tm=no
with_devel_headers=yes
with_portals=no
with_valgrind=no
```

**Comment by nrgraham23** [comment]
Ill verify I get the warnings with 1.10, and if so, send a PR.

**Comment by nrgraham23** [comment]
v1.10 is not picking up my java installation.  What do I need to do to make it find it?

**Comment by rhc54** [comment]
I honestly don't know - haven't done anything special myself other than --enable-mpi-java. I have the following java-related packages installed on my CentOS-7 box:

```
java-1.8.0-openjdk.x86_64              1:1.8.0.101-3.b13.el7_2         @updates 
java-1.8.0-openjdk-devel.x86_64        1:1.8.0.101-3.b13.el7_2         @updates 
java-1.8.0-openjdk-headless.x86_64     1:1.8.0.101-3.b13.el7_2         @updates 
javamail.noarch                        1.4.6-8.el7                     @anaconda
javapackages-tools.noarch              3.4.1-11.el7                    @anaconda
javassist.noarch                       3.16.1-10.el7                   @anaconda
libvirt-java.noarch                    0.4.9-4.el7                     @anaconda
libvirt-java-devel.noarch              0.4.9-4.el7                     @anaconda
nuxwdog-client-java.x86_64             1.0.3-4.el7_2                   @updates 
python-javapackages.noarch             3.4.1-11.el7                    @anaconda
tzdata-java.noarch                     2016f-1.el7                     @updates 
```

**Comment by nrgraham23** [comment]
I was able to get it to work on one of our linux machines here.  It must be an issue related to Mac, but Ill wait until @hppritcha can take a look at it before opening an issue.  

I saw the warnings you were talking about, and it does look like what I fixed almost a year ago in master.  Hopefully there are not so many conflicts with a cherry-pick that its faster to simply redo the work, but Ill get a PR up here in the next few hours.
